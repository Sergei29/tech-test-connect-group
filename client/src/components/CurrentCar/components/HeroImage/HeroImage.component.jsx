import React from "react";
import PropTypes from "prop-types";
//style:
import { HeroImageContainer } from "./HeroImage.style";

/**
 * @description functiona lcomponent for Hero Image display
 * @param {String} {imgUrl path ot image
 * @param {String} headingText car make, model
 * @param {String} subText} car short description
 * @returns {JSX} component markup
 */
const HeroImage = ({ imgUrl, headingText, subText }) => {
  return (
    <HeroImageContainer imgUrl={imgUrl}>
      <div className="heroText">
        <h1>{headingText}</h1>
        <p className="subText">{subText}</p>
      </div>
    </HeroImageContainer>
  );
};

HeroImage.defaultProps = {
  headingText: "",
  subText: "",
};

HeroImage.propTypes = {
  imgUrl: PropTypes.string.isRequired,
  headingText: PropTypes.string,
  subText: PropTypes.string,
};

export default HeroImage;
