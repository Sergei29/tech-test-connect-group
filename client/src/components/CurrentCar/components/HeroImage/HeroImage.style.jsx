import styled from "styled-components";
import { mediaQueryTo } from "../../../../constants/styleMixins";
import { spacing, color } from "../../../../constants/styleVariables";

export const HeroImageContainer = styled.div`
  background-image: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)),
    url(${(props) => props.imgUrl});
  height: 65vh;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  position: relative;
  ${mediaQueryTo.md`
    height: 60vh;
    background-position: bottom;
  `}
  ${mediaQueryTo.sm`
    background-position: bottom;
  `}
  ${mediaQueryTo.xs`
    height: 40vh;
  `}
  .hero-text {
    text-align: center;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    color: ${color.white};
    h1 {
      font-size: 50px;
      margin-bottom: ${spacing.unit(4)};
      ${mediaQueryTo.sm`
        font-size: 34px;
      `};
    }
    button {
      border: none;
      outline: 0;
      display: inline-block;
      padding: ${spacing.unit(1)} ${spacing.unit(3)};
      color: ${color.black};
      background-color: ${color.greyLight};
      text-align: center;
      cursor: pointer;
      &:hover {
        background-color: ${color.greyDark};
        color: ${color.white};
      }
    }
  }
`;
