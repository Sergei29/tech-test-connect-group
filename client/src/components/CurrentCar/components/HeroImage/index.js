export { default } from "./HeroImage.component";
