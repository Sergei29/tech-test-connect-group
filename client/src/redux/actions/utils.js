/**
 * @description generates the error message out of received error from api
 * @param {Object} error error received from api
 * @param {String} entity name of the entity: 'carList' or 'carCurrent'
 * @returns {String} aggregated error message
 */
export const getErrorMessage = (error, entity) => {
  let errorMessage = "API call failed.";

  if (error.response && error.response.data) {
    errorMessage = error.response.data;
  } else if (error.message) {
    errorMessage = error.message;
  }

  return `${errorMessage}. ${entity}`;
};
