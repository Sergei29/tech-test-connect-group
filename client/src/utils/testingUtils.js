import React from "react";
import { shallow, mount } from "enzyme";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

const mockStore = configureMockStore([thunk]);

const store = mockStore({
  carList: [
    {
      id: "xe",
      modelYear: "k17",
      url: "/vehicle/xe",
      description:
        "The most advanced, efficient and refined sports saloon that Jaguar has ever produced",
      price: "£30,000",
      media: [{ media: "vehicle", url: "assets/images/xe_k17.jpg" }],
    },
    {
      id: "xf",
      modelYear: "k17",
      url: "/vehicle/xf",
      description:
        "Luxury business saloon with distinctive design, dynamic drive and state-of-the-art technologies.",
      price: "£36,000",
      media: [{ media: "vehicle", url: "assets/images/xf_k17.jpg" }],
    },
  ],
  carCurrent: {
    id: "xe",
    meta: {
      passengers: 5,
      drivetrain: ["AWD", "RWD"],
      bodystyles: ["saloon"],
      emissions: { template: "CO2 Emissions $value g/km", value: 99 },
    },
  },
  error: null,
  loading: false,
});

export const shallowRendererReduxConnected = (ComponentName, props = {}) =>
  shallow(
    <Provider store={store}>
      <BrowserRouter>
        <ComponentName {...props} />
      </BrowserRouter>
    </Provider>
  );

export const mountRendererReduxConnected = (ComponentName, props = {}) =>
  mount(
    <Provider store={store}>
      <BrowserRouter>
        <ComponentName {...props} />
      </BrowserRouter>
    </Provider>
  );

export const shallowRenderer = (ComponentName, props = {}) =>
  shallow(<ComponentName {...props} />);

export const mountRenderer = (ComponentName, props = {}) =>
  mount(
    <BrowserRouter>
      <ComponentName {...props} />
    </BrowserRouter>
  );
