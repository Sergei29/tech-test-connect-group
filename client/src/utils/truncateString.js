export const truncateString = (stringValue, maxLength) => {
  let truncateLocation = stringValue.indexOf(" ", maxLength);
  if (truncateLocation === -1) return stringValue;
  return stringValue.substring(0, truncateLocation) + "...";
};
