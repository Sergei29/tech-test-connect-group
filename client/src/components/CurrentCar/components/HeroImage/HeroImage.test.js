import { shallowRenderer } from "../../../../utils/testingUtils";
import HeroImage from "./HeroImage.component";

describe("HeroImage", () => {
  let wrapper, props;

  beforeEach(() => {
    props = {
      headingText: "Jaguar k17 xe",
      imgUrl: "/assets/images/xe_k17.jpg",
      subText:
        "The most advanced, efficient and refined sports saloon that Jaguar has ever produced",
    };
    wrapper = shallowRenderer(HeroImage, props);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
