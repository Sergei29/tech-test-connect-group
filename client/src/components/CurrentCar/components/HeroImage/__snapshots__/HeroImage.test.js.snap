// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroImage should match the snapshot 1`] = `
<styled.div
  imgUrl="/assets/images/xe_k17.jpg"
>
  <div
    className="heroText"
  >
    <h1>
      Jaguar k17 xe
    </h1>
    <p
      className="subText"
    >
      The most advanced, efficient and refined sports saloon that Jaguar has ever produced
    </p>
  </div>
</styled.div>
`;
